primitives/src/asset.rs

```diff
@@ -62,6 +62,10 @@ pub enum AssetType {
     Derivative,
     /// Anything else.
     Custom(Vec<u8>),
+    /// Stablecoins are cryptocurrencies designed to minimize the volatility of the price of the stablecoin,
+    /// relative to some "stable" asset or basket of assets.
+    /// A stablecoin can be pegged to a cryptocurrency, fiat money, or to exchange-traded commodities.
+    StableCoin,
```

Comment:
Seems it would be helpful having a general design:
```rust
enum OrCustom<A, B = Vec<u8>> {
    Custom(B),
    Builtin(A),
}

impl<A: Default, B> Default for OrCustom<A, B> { ... }
```
Here `AssetType` becomes `OrCustom<BultinAssetType>` and you'll have e.g., `OrCustom<BuiltinAssetType, CustomAssetTypeId>` for the "custom info is stored in a different storage item" case to ensure we make the primitive case cheaper.

This idea would also be used for external agents.